src/components/Modal/index.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ConfirmationModal from './confirmationModal';
+import JSONModal from './jsonModal';
+import './style.scss';
+
+export default class Modal extends React.Component {
+  render() {
+    const {
+      opened, type, onChange, title, message, submit, cancel, attendees,
+    } = this.props;
+    if (opened) {
+      if (type === 'confirmation') {
+        return (
+          <ConfirmationModal
+            onChange={onChange}
+            title={title}
+            message={message}
+            submit={submit}
+            cancel={cancel}
+          />
+        );
+      }
+      if (type === 'json') {
+        return (
+          <JSONModal
+            onChange={onChange}
+            title={title}
+            attendees={attendees}
+          />
+        );
+      }
+    }
+    return null;
+  }
+}
+
+Modal.defaultProps = {
+  submit: undefined,
+  cancel: undefined,
+  attendees: undefined,
+  message: undefined,
+};
+
+Modal.propTypes = {
+  opened: PropTypes.bool.isRequired,
+  type: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
```

Comment:
What is this function supposed to do?

Reply:
This is correct. React has a built in way of defining annotations for component props and state. You assign a javascript object to `Modal.propTypes` that specifies which properties are valid, what type they are (or a custom validator), and whether they are required. It's for documentation and type-checking purposes. The react compiler can then deduce if you're passing an invalid prop anywhere and let you know.

`Modal.defaultProps` is similar except instead of a type, you provide a default value for each optional prop.